okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -515,6 +516,16 @@ public void proxy(RoutingContext ctx) {
     }
 
     final MultiMap headers = ctx.request().headers();
+
+    FolioLoggingContext.put(FolioLoggingContext.TENANT_ID_LOGGING_VAR_NAME,
+        tenantId);
+    FolioLoggingContext.put(FolioLoggingContext.REQUEST_ID_LOGGING_VAR_NAME,
+        headers.get(XOkapiHeaders.REQUEST_ID));
+    FolioLoggingContext.put(FolioLoggingContext.MODULE_ID_LOGGING_VAR_NAME,
+        headers.get(XOkapiHeaders.MODULE_ID));
+    FolioLoggingContext.put(FolioLoggingContext.USER_ID_LOGGING_VAR_NAME,
+        headers.get(XOkapiHeaders.USER_ID));
```

Comment:
When running `mvn clean install` there isn't a single unit test that logs a user id. Can you extend an existing test by adding XOkapiHeaders.USER_ID? You don't need to assert that it is logged, I can look at the output.

Reply:
done

Reply:
done

Reply:
This works if the user id is passed in as X-Okapi-User-Id header.
However, the front-end passes in the user id in the token. Okapi sends the token to mod-authtoken for validation. When Okapi gets the reply from mod-authtoken that the validation was successful we know the user id and can set the user id in the FolioLoggingContext. Details: https://github.com/folio-org/okapi/blob/master/doc/security.md
Can you find the code where Okapi gets the validation result and add code for setting the user id?

Reply:
Are we interested in userId, username (login) or both? I searched Okapi code and looks like currently it does not have a central place to extract those info from the token. I am thinking that we can add those methods to https://github.com/folio-org/okapi/blob/master/okapi-common/src/main/java/org/folio/okapi/common/OkapiToken.java. For example like this https://github.com/folio-org/okapi/compare/username-userid?expand=1